aana/deployments/speaker_diarization_deployment.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, TypedDict
+
+import torch
+from pyannote.audio import Pipeline
+from pyannote.core import Annotation
+from pydantic import BaseModel, ConfigDict, Field
+from ray import serve
+
+from aana.core.models.audio import Audio
+from aana.core.models.base import pydantic_protected_fields
+from aana.core.models.speaker import SpeakerDiarizationParams, SpeakerDiarizationSegment
+from aana.core.models.time import TimeInterval
+from aana.deployments.base_deployment import BaseDeployment
+from aana.exceptions.runtime import InferenceException
+
+
+class SpeakerDiarizationOutput(TypedDict):
+    """The output of the Speaker Diarization model.
+
+    Attributes:
+        segments (list[SpeakerDiarizationSegment]): The Speaker Diarization segments.
+    """
+
+    segments: list[SpeakerDiarizationSegment]
+
+
+class SpeakerDiarizationConfig(BaseModel):
+    """The configuration for the Speaker Diarization deployment.
+
+    Attributes:
+        model_name (str): name of the speaker diarization pipeline.
+        sample_rate (int): The sample rate of the audio. Defaults to 16000.
+    """
+
+    model_name: str = Field(
+        default="pyannote/speaker-diarization-3.1",
+        description="The Speaker Diarization model name.",
+    )
+
+    sample_rate: int = Field(default=16000, description="Sample rate of the audio.")
+
+    model_config = ConfigDict(protected_namespaces=(*pydantic_protected_fields,))
+
+
+@serve.deployment
+class SpeakerDiarizationDeployment(BaseDeployment):
```

Comment:
Do we plan to support other diarization libraries? This deployment is specific to pyannotate. So it would be more precise to call it PyannotateSpeakerDiarizationDeployment.

Reply:
We do not have a competing and simple library at the moment. The idea is to rename it if something else comes in the future. Else, we could use this as the default `SpeakerDiarizationDeployment`. This is my opinion, open to discussion.

Reply:
If you think we might rename it in the future, it's better to rename it now because we will have applications that use it and we will have to rename it everywhere.

Reply:
It is possible that it will be renamed if we get a better model in the long term. By this naming convention, WhisperDeployment needs to be `WhisperSpeechRecognitionDeployment` isn't it?

Reply:
Renamed the deployment and associated variables to conform to the deployment name:`PyannotateSpeakerDiarizationDeployment`